test/spec/modules/ttdBidAdapter_spec.js

```diff
@@ -0,0 +1,1090 @@
+        bid.params.categories = 'IAB23';
+        expect(spec.isBidRequestValid(bid)).to.equal(false);
+      });
+
+      it('should return false if categories is not passed as an array of strings', function () {
+        let bid = makeBid();
+        bid.params.categories = [233];
+        expect(spec.isBidRequestValid(bid)).to.equal(false);
+      });
+
+      it('should return false if categories is not passed as an array of strings', function () {
+        let bid = makeBid();
+        bid.params.categories = [233];
+        expect(spec.isBidRequestValid(bid)).to.equal(false);
+      });
+    });
+
+    describe('banner', function () {
+      it('should return true if banner.pos is passed correctly', function () {
+        let bid = makeBid();
+        bid.mediaTypes.banner.pos = 1;
+        expect(spec.isBidRequestValid(bid)).to.equal(true);
+      });
+
+      it('should return true if banner.expdir is passed correctly', function () {
+        let bid = makeBid();
+        bid.params.banner = {
+          expdir: [1, 3]
+        };
+        expect(spec.isBidRequestValid(bid)).to.equal(true);
+      });
+
+      it('should return false if banner.expdir is not passed as an array', function () {
+        let bid = makeBid();
+        bid.params.banner = {
+          expdir: 1
+        };
+        expect(spec.isBidRequestValid(bid)).to.equal(false);
+      });
+    });
+
+    describe('video', function () {
+      function makeBid() {
+        return {
+          'bidder': 'ttd',
+          'params': {
+            'supplySourceId': 'supplier',
+            'publisherId': '22222222',
+            'siteId': 'testSiteId123',
+            'placementId': 'somePlacementId'
+          },
+          'mediaTypes': {
+            'video': {
+              'minduration': 5,
+              'maxduration': 30,
+              'playerSize': [640, 480],
+              'api': [1, 3],
+              'mimes': ['video/mp4'],
+              'protocols': [2, 3, 5, 6]
+            }
+          },
+          'adUnitCode': 'adunit-code',
+          'sizes': [
+            [300, 250]
+          ],
+          'bidId': '30b31c1838de1e',
+          'bidderRequestId': '22edbae2733bf6',
+          'auctionId': '1d1a030790a475',
+        };
+      }
+
+      it('should return true if required parameters are passed', function () {
+        let bid = makeBid();
+        expect(spec.isBidRequestValid(bid)).to.equal(true);
+      });
+
+      it('should return false if minduration is missing', function () {
+        let bid = makeBid();
+        delete bid.mediaTypes.video.minduration;
+        expect(spec.isBidRequestValid(bid)).to.equal(false);
+      });
+
+      it('should return false if maxduration is missing', function () {
+        let bid = makeBid();
+        delete bid.mediaTypes.video.maxduration;
+        expect(spec.isBidRequestValid(bid)).to.equal(false);
+      });
+
+      it('should return false if playerSize is missing', function () {
```

Comment:
why not attempt to construct from w & h if missing

Reply:
after reviewing our server side bidder, i realized video.w and video.h aren't required in the bid request, so i've removed playerSize as a required parameter